File: code-examples/javascript/09/01/test/LoginTest.js
const { Builder } = require('selenium-webdriver')
const { Eyes } = require('@applitools/eyes-selenium')
const path = require('path')
const assert = require('assert')
const LoginPage = require('../pages/LoginPage')

describe('Login', function() {
  this.timeout(30000)
  let driver
  let eyes
  let login

  beforeEach(async function() {
    const vendorDirectory =
      path.delimiter + path.join(__dirname, '..', 'vendor')
    process.env.PATH += vendorDirectory
    driver = await new Builder().forBrowser('firefox').build()
    const hasEyesCommands = this.currentTest.body.match(/eyes\./)
    if (hasEyesCommands) {
      eyes = new Eyes()
      eyes.setApiKey(process.env.APPLITOOLS_API_KEY)
      driver = await eyes.open(
        driver,
        'the-internet',
        this.currentTest.fullTitle(),
        {
          width: 1024,
          height: 768,
        }
      )
    }
    login = new LoginPage(driver)
    await login.load()
  })

  afterEach(async function() {
    await driver.quit()
    if (eyes) await eyes.abortIfNotClosed()
  })

  it('with valid credentials', async function() {
    await login.authenticate('tomsmith', 'SuperSecretPassword!')
    assert(await login.successMessagePresent(), 'Success message not displayed')
    await eyes.checkWindow('Logged in')
    await eyes.close()
  })
})

File: code-examples/python/09/01/tests/login_test.py
import pytest
import os
from selenium import webdriver
from pages import login_page


class TestLogin():

    @pytest.fixture
    def login(self, request):
        _geckodriver = os.path.join(os.getcwd(), 'vendor', 'geckodriver')
        if os.path.isfile(_geckodriver):
            driver_ = webdriver.Firefox(executable_path=_geckodriver)
        else:
            driver_ = webdriver.Firefox()

        def quit():
            driver_.quit()

        request.addfinalizer(quit)
        return login_page.LoginPage(driver_)

    def test_valid_credentials(self, login):
        login.with_("tomsmith", "SuperSecretPassword!")
        assert(login.success_message_present())

    def test_invalid_credentials(self, login):
        login.with_("tomsmith", "bad password")
        assert(login.failure_message_present())
